### PR TITLE
Oxidation parsec test macro

### LINKED ISSUE #4267: Simplify rust test by having our own #[test] proc macro
Currently the tests in rust:
```rust
    #[rstest] // 1. Rstest
    #[test_log::test(tokio::test)] // 2. tokio + log catcher
    async fn test_chunk_many() {
        TestbedScope::run("minimal", |env| async move {  // 3. testbed wrapper
            let alice = env.local_device("alice@dev1".parse().unwrap());
            let aws = workspace_storage_with_defaults(&env.client_config_dir, alice, None).await;
            // actual test code
        })
        .await;
    }
```
It would be much better to have our own `#[test]` proc macro to simplify this:
```rust
    #[parsec_test(testbed="minimal")]
    async fn test_chunk_many(env: TestbedEnv) {
        let alice = env.local_device("alice@dev1".parse().unwrap());
        let aws = workspace_storage_with_defaults(&env.client_config_dir, alice, None).await;
        // actual test code
}
```

### PR DESCRIPTION
**Pull request by TimeEngineer**
# What has changed ?

- [ ] Does this PR affect the User ?

closes #4267

### CODE REVIEW

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]
I don't think we should use the crate `libparsec_parsec_test` where we don't use the testbed.

**Comment by touilleMan** [comment]
image

**Comment by TimeEngineer** [comment]
Why ? It's independant

**Review by touilleMan — CHANGES_REQUESTED** [review]

File: oxidation/libparsec/crates/client_connection/tests/auth_cmds.rs [hunk 1]
```diff
@@ -176,32 +177,29 @@ async fn client(
...
-            rep.unwrap(),
-            libparsec_protocol::authenticated_cmds::v3::ping::Rep::Ok {
-                pong: "foo".to_owned()
-            }
-        );
-    })
-    .await
+#[parsec_test(testbed = "coolorg", with_server)]
+async fn with_testbed(env: &TestbedEnv) {
+    let client = reqwest::ClientBuilder::new()
```
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply oxidation/libparsec/crates/client_connection/tests/auth_cmds.rs:182]: Nothing more

**Comment by TimeEngineer** [comment]
> > Why ? It's independant
> 
> > **Smaller is better**. You should include the minimal set of changes required
> > to address the subject of the pull request.

The PR address to use `parsec_test` instead of all `test`, you can have 2 PRs for that but changing 1 line is not the big part of this PR

File: oxidation/libparsec/crates/client_connection/Cargo.toml [hunk 2]
```diff
@@ -33,12 +33,13 @@ chrono = "0.4.23"
 thiserror = "1.0.38"
 
 [dev-dependencies]
+libparsec_parsec_test = { path = "../parsec_test" }
```
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply oxidation/libparsec/crates/client_connection/Cargo.toml]: Done

File: oxidation/libparsec/crates/parsec_test/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+edition = "2021"
+license = " BUSL-1.1"
+
+[lib]
+proc-macro = true
+
+[dependencies]
+proc-macro2 = "1.0.51"
+quote = "1.0.23"
+syn = { version = "1.0.109" }
```
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply oxidation/libparsec/crates/parsec_test/Cargo.toml]: Current Branch (all):
```
real    1m54,068s
user    18m11,588s
sys     1m13,581s
```

Master (all):
```
real    1m23,605s
user    17m25,681s
sys     1m11,931s
```

Current Branch (core_fs):
```
real    0m7,708s
user    0m14,242s
sys     0m1,494s
```

Master (core_fs):
```
real    0m7,696s
user    0m14,201s
sys     0m1,568s
```

File: oxidation/libparsec/crates/tests_macros/src/lib.rs
Comment (FirelightFlagboy) [inline oxidation/libparsec/crates/tests_macros/src/lib.rs:1]: Can you add `with_log` ?

**Review by FirelightFlagboy — COMMENTED** [review]
With that PR the amount of cyclic dependencies is increase on the test level, can you confirm that rust-analyzer is working properly ?

File: oxidation/libparsec/crates/tests_macros/src/lib.rs
Comment (FirelightFlagboy) [reply oxidation/libparsec/crates/tests_macros/src/lib.rs:1]: Currently, we can't enable the log in the test, we lack the use of `test_log`.

I propose to add `with_log` that way we only enable for test that need it.

Comment (TimeEngineer) [reply oxidation/libparsec/crates/tests_macros/src/lib.rs:1]: `test_log` macro is removed because my macro initialize the `logger`

**Comment by TimeEngineer** [comment]
> With that PR the amount of cyclic dependencies is increase on the test level, can you confirm that rust-analyzer is working properly ?

It won't work because `tests_fixtures` broke it with testbed :stuck_out_tongue:

**Review by FirelightFlagboy — APPROVED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**